Python/bytecodes.c

```diff
@@ -2286,38 +2286,40 @@ dummy_func(
         }
 
         op(_LOAD_ATTR_WITH_HINT, (hint/1, owner, dict: PyDictObject * -- attr)) {
+            PyDictKeysObject *dk = FT_ATOMIC_LOAD_PTR(dict->ma_keys);
```

Comment:
I think that we need to check the equivalent of `ensure_shared_on_read` either here or in `_CHECK_ATTR_WITH_HINT`.